src/test/rgw/test_rgw_lua.cc

```diff
@@ -280,6 +280,121 @@ TEST(TestRGWLua, URI)
   ASSERT_EQ(rc, 0);
 }
 
+TEST(TestRGWLua, RGWDebugLogF)
+{
+  const std::string script = R"(
+    RGWDebugLogF("Testing {} {} {} {}", "hello", 1234567890, true, nil)
+  )";
+
+  DEFINE_REQ_STATE;
+
+  const auto rc = lua::request::execute(nullptr, nullptr, &s, nullptr, script);
+  ASSERT_EQ(rc, 0);
+}
+
+TEST(TestRGWLua, RGWDebugLogFTooManyArguments)
+{
+  const std::string script = R"(
+    RGWDebugLogF("Testing {}", "hello", 1234567890)
+  )";
+
+  DEFINE_REQ_STATE;
+
+  const auto rc = lua::request::execute(nullptr, nullptr, &s, nullptr, script);
+  ASSERT_EQ(rc, 0);
+}
+
+TEST(TestRGWLua, RGWDebugLogFTooFewArguments)
+{
+  const std::string script = R"(
+    RGWDebugLogF("Testing {} , {}", 1234567890)
+  )";
+
+  DEFINE_REQ_STATE;
+
+  const auto rc = lua::request::execute(nullptr, nullptr, &s, nullptr, script);
+  ASSERT_EQ(rc, 0);
+}
+
+TEST(TestRGWLua, RGWDebugLogFNoArguments)
+{
+  const std::string script = R"(
+    RGWDebugLogF("Hello World")
+  )";
+
+  DEFINE_REQ_STATE;
+
+  const auto rc = lua::request::execute(nullptr, nullptr, &s, nullptr, script);
+  ASSERT_EQ(rc, 0);
+}
+
+TEST(TestRGWLua, RGWDebugLogFUnsupportedFormat)
+{
+  const std::string script = R"(
+    RGWDebugLogF("Testing {d}, {{}} {1}", 1234567890, true)
+  )";
+
+  DEFINE_REQ_STATE;
+
+  const auto rc = lua::request::execute(nullptr, nullptr, &s, nullptr, script);
+  ASSERT_EQ(rc, 0);
+}
+
+TEST(TestRGWLua, RGWDebugLogFInvalidFormat)
+{
+  const std::string script = R"(
+    RGWDebugLogF("Testing {} incomplete format string {", 1234567890, true)
+  )";
+
+  DEFINE_REQ_STATE;
+
+  const auto rc = lua::request::execute(nullptr, nullptr, &s, nullptr, script);
+  ASSERT_EQ(rc, 0);
+}
+
+TEST(TestRGWLua, RGWDebugLogFUnsupportedType)
+{
+  const std::string script = R"(
+    RGWDebugLogF("Testing table {}", {"abc", "def"})
+  )";
+
+  DEFINE_REQ_STATE;
+
+  const auto rc = lua::request::execute(nullptr, nullptr, &s, nullptr, script);
+  ASSERT_EQ(rc, 0);
+}
+
+TEST(TestRGWLua, RGWDebugLogFInvalidField)
+{
+  const std::string script = R"(
+    RGWDebugLogF("Testing {}", Request.Kaboom)
+  )";
+
+  DEFINE_REQ_STATE;
+
+  const auto rc = lua::request::execute(nullptr, nullptr, &s, nullptr, script);
+  ASSERT_EQ(rc, -1);
+}
+
+TEST(TestRGWLua, RGWDebugLogFRequest)
+{
+  const std::string script = R"(
+    RGWDebugLogF("Testing {}", Request.Bucket)
```

Comment:
i think that this one would fail as `Request.Bucket` is a table.
maybe also test something like: `Request.Bucket.Name`